src/common/snippets/src/pass/transpose_decomposition.cpp

```diff
@@ -0,0 +1,115 @@
+                for (size_t w = 0; w < W; w++) {
+                    for (size_t c = 0; c < C; c++) {
+                        // src_ptr[n*H*W*C + h*W + c*H*W + w]
+                        *dst_ptr = *src_ptr;
+                        dst_ptr += data_size; // it's easy with dst, since scheduling is dst-dense => simply increment by data size
+                        src_ptr += H*W * data_size; // access to src is HW strided
+                    }
+                    src_ptr += (1 - H*W*C) * data_size; // step up along the W dimension
+                }
+                // src_inner_ptr is already incremented by W at this point, HW is dense => do nothing
+            }
+            // src_inner_ptr is already incremented by H*W at this point => increment by (C-1)*H*W
+            src_inner_ptr += (C-1)*H*W;
+        }
+         */
+        auto data_input = pattern_to_output.at(match_data);
+        auto &param_rt = data_input.get_node_shared_ptr()->get_rt_info();
+        // Note: store and usage inside emitters as size_t is more convenient, so static_cast here
+        std::vector<size_t> access_pattern;
+        std::copy(order_value.begin(), order_value.end(), std::back_inserter(access_pattern));
+        param_rt["NonDefaultAccessPattern"] = access_pattern;
+
+        // The line below is Ok, since we ensured that transpose is static above
+        auto data_shape = data_input.get_shape();
+        // dim indexes with respect to SRC
+        const auto dim_C_idx = data_shape.size() - 3;
+        const auto dim_H_idx = data_shape.size() - 2;
+        const auto dim_W_idx = data_shape.size() - 1;
+        const auto size_C = static_cast<int64_t>(data_shape[dim_C_idx]);
+        const auto size_W = static_cast<int64_t>(data_shape[dim_W_idx]);
+        const auto size_H = static_cast<int64_t>(data_shape[dim_H_idx]);
+
+        auto loop_W_begin = std::make_shared<op::LoopBegin>(OutputVector{data_input});
+        auto loop_C_begin = std::make_shared<op::LoopBegin>(OutputVector{loop_W_begin->output(0)});
+        // todo: LoadReshape used here is essentially Load + an easy way to maintain correct shape propagation
+        //  fix this in future and develop a more consistent shape propagation approach.
+        auto load = std::make_shared<snippets::op::LoadReshape>(loop_C_begin->output(0), 1, access_pattern);
+        auto store = std::make_shared<snippets::op::Store>(load->output(0), 1);
+        // todo: extend Loop functionality: provide method to override default pointer increments:
+        //  default: pointer_increment = WA_increment * data_size
+        //  extended: pointer_increment >= WA_increment * data_size for strided data access
+        // work around with finalization offset for now, but this is an additional addition!
+        const std::vector<int64_t> ptr_increments_C {size_H * size_W, 1};
+        const std::vector<int64_t> finalization_offsets_C {1 - size_H * size_W * size_C, 0};
```

Comment:
Need discuss case when input/output is `Buffer`

Reply:
I suggest we implement transpose + eltwise in a separate PR. what do you think?